Source/WebKit/UIProcess/RemoteLayerTree/mac/ScrollingTreeFrameScrollingNodeRemoteMac.cpp

```diff
@@ -53,12 +53,20 @@ void ScrollingTreeFrameScrollingNodeRemoteMac::commitStateBeforeChildren(const S
 {
     ScrollingTreeFrameScrollingNodeMac::commitStateBeforeChildren(stateNode);
     const auto& scrollingStateNode = downcast<ScrollingStateFrameScrollingNode>(stateNode);
-    CALayer* horizontalLayer = nullptr;
-    CALayer* verticalLayer = nullptr;
-    m_delegate->getScrollbarLayersForStateNode(scrollingStateNode, &horizontalLayer, &verticalLayer);
-    m_scrollerPair->horizontalScroller().setHostLayer(horizontalLayer);
-    m_scrollerPair->verticalScroller().setHostLayer(verticalLayer);
     
+    if (scrollingStateNode.hasChangedProperty(ScrollingStateNode::Property::VerticalScrollbarLayer) && horizontalNativeScrollbarVisibility() == NativeScrollbarVisibility::Visible)
```

Comment:
I'm not convinced this is better. There's more code now in 2 places.

Reply:
Current plan is to move the scroller pair to ScrollingTreeScrollingNodeDelegateMac and share this code there.